router/router_test.go

```diff
@@ -1350,6 +1350,50 @@ var _ = Describe("Router", func() {
 
 	})
 
+	Context("serving multiple http versions", func() {
```

Comment:
This context says "serving multiple http versions" but then the only test is about HTTP/1.1 and I don't see any references to other versions.

This the fact that the h2 test is missing related to your comment in the PR: 
> We were not able to get a router/router test working because making
HTTP/2 requests to the test Server resulted in broken pipes. We believe
this is due to differences in the test Server setup and the main.go
setup used in the integration tests. This will require additional
investigation.

❓ So are you missing test coverage here in order to keep the tests passing? 
❓ If so, can you add the test even if it is broken so others can more easily look into it?
❓ Can you also add more details about how the test is failing?

Reply:
The challenge that we ran into when trying to write the test is that the test setup for router_test.go doesn't work with h2 as the test setup is not using the production setup code and it wasn't clear how to change it. This is probably because h2 requires TLS.

We added test coverage for all these cases in the integration tests, so I think it makes sense to just delete this one loner test.